From eb21cdeb6d0493037aba9a2a356dfa9c7a611179 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:04 +0800
Subject: [PATCH 046/150] dt-bindings: phytium-sgpio: Add bindings for Phytium
 SGPIO

This patch documents the DT bindings for Phytium SGPIO controller.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: Ie97e2b000b0a8489268eb1863f69c15005e05be0
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../bindings/gpio/phytium,sgpio.yaml          | 67 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 68 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/gpio/phytium,sgpio.yaml

diff --git a/Documentation/devicetree/bindings/gpio/phytium,sgpio.yaml b/Documentation/devicetree/bindings/gpio/phytium,sgpio.yaml
new file mode 100644
index 000000000000..fed5c2b63468
--- /dev/null
+++ b/Documentation/devicetree/bindings/gpio/phytium,sgpio.yaml
@@ -0,0 +1,67 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/gpio/phytium,sgpio.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium SGPIO controller
+
+description: |
+  This SGPIO controller is for Phytium Pe220x SoCs, which supports up to
+  96 (32x3) Serial GPIOs.
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+properties:
+  compatible:
+    const: phytium,sgpio
+
+  reg:
+    maxItems: 1
+    description: Address and length of the register set for the device.
+
+  gpio-controller: true
+
+  '#gpio-cells':
+    const: 2
+    description: |
+      The first cell is the pin number and the second cell is used to specify
+      the gpio polarity.
+        0 = active high
+        1 = active low
+
+  interrupts:
+    maxItems: 1
+
+  ngpios: true
+
+  bus-frequency: true
+
+  clocks:
+    maxItems: 1
+
+additionalProperties: false
+
+required:
+  - compatible
+  - reg
+  - gpio-controller
+  - '#gpio-cells'
+  - interrupts
+  - ngpios
+  - clocks
+  - bus-frequency
+
+examples:
+  - |
+    sgpio: sgpio@2807d000 {
+      compatible = "phytium,sgpio";
+      reg = <0x0 0x2807d000 0x0 0x1000>;
+      interrupts = <GIC_SPI 190 IRQ_TYPE_LEVEL_HIGH>;
+      clocks = <&sysclk_48mhz>;
+      ngpios = <96>;
+      bus-frequency = <48000>;
+      gpio-controller;
+      #gpio-cells = <2>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 99300c36e2da..1f0e836c6717 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2531,6 +2531,7 @@ M:	Chen Baozi <chenbaozi@phytium.com.cn>
 S:	Maintained
 W:	https://www.phytium.com.cn
 F:	Documentation/devicetree/bindings/gpio/phytium,gpio.yaml
+F:	Documentation/devicetree/bindings/gpio/phytium,sgpio.yaml
 F:	Documentation/devicetree/bindings/hwlock/phytium,hwspinlock.yaml
 F:	Documentation/devicetree/bindings/i2c/phytium,i2c.yaml
 F:	Documentation/devicetree/bindings/media/phytium,jpeg.yaml
-- 
2.47.0

